### PR TITLE
[e2e] Move framework/upgrade_util.go to framework/lifecycle/upgrade.go

### PR DESCRIPTION
**Pull request by jiatongw**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:

- Move upgrade_util.go to framework/cluster.
- Fix references of some methods to use the right package

Reference issue #76206 (parent issue #75601 )

**Special notes for your reviewer**:
Reason for moving this to `cluster` package: All of the functions in this file are called by `e2e/lifecycle/cluster_upgrade.go`, and there is already a package named `upgrades`( `e2e/upgrades`). So I avoid using `upgrade(s)`. Also regarding that `cluster` is a generic naming that not such "big", and new tests related with `cluster` can be also added here. 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
thanks @jiatongw 

/kind cleanup
/ok-to-test
/priority backlog

> Reason for moving this to cluster package: All of the functions in this file are called by e2e/lifecycle/cluster_upgrade.go, and there is already a package named upgrades( e2e/upgrades). So I avoid using upgrade(s).

hm, i'm not sure about using cluster as the name.
we probably should go with framework/upgrade, but let's see what the others think.

also, we might want to stop supporting the upgrade utils in the framework.
we need to evaluate which test jobs use test/e2e/lifecycle/cluster_upgrade.go nowadays.
one would be:
https://k8s-testgrid.appspot.com/sig-release-master-upgrade#gce-new-master-upgrade-master-parallel

**Comment by jiatongw** [comment]
Thanks @neolit123 . I'll also look up the test jobs that use `e2e/lifecycle/cluster_upgrade.go`

**Comment by jiatongw** [comment]
@andrewsykim what is your opinion?

**Comment by andrewsykim** [comment]
Putting upgrade related utilities under `test/e2e/framework/lifecycle` makes sense to me.

**Comment by jiatongw** [comment]
> Moving upgrade related utilities and duplicated code in `test/e2e/lifecycle/` to `test/e2e/framework/lifecycle` (over `test/e2e/framework/cluster`) makes sense to me.

There is a potential new file `versions.go` in lifecycle package. 
```
lifecycle/ 
- versions.go
```
Either move all the code in `upgrade_util.go` to `versions.go`, or just move `upgrade_util.go` to lifecycle package works for me. How about you?

**Comment by andrewsykim** [comment]
How about just `test/e2e/framework/lifecycle/upgrade.go`?

**Comment by jiatongw** [comment]
> How about `test/e2e/framework/lifecycle/upgrade.go`?

👍 SGTM

**Comment by jiatongw** [comment]
@andrewsykim able to approve?

File: test/e2e/framework/lifecycle/upgrade.go [hunk 1]
```diff
@@ -26,39 +26,40 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apimachinery/pkg/version"
 	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
```
Comment (andrewsykim) [inline test/e2e/framework/lifecycle/upgrade.go]: Can you use `Logf` from `test/e2e/framework/log`?

Comment (jiatongw) [reply test/e2e/framework/lifecycle/upgrade.go]: Yup just see your issue. I'll fix that. Thanks

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: test/e2e/framework/lifecycle/upgrade.go (hunk 1 above)
Comment (jiatongw) [reply test/e2e/framework/lifecycle/upgrade.go]: Fixed

File: test/e2e/framework/lifecycle/upgrade.go [hunk 2]
```diff
@@ -26,39 +26,41 @@ import (
...
 	if exiterr, ok := err.(*exec.ExitError); err != nil && ok {
-		Logf("Error while running traceroute: %s", exiterr.Stderr)
+		e2elog.Logf("Error while running traceroute: %s", exiterr.Stderr)
 	}
 }
 
 // CheckMasterVersion validates the master version
 func CheckMasterVersion(c clientset.Interface, want string) error {
-	Logf("Checking master version")
+	framework.Logf("Checking master version")
```
Comment (andrewsykim) [inline test/e2e/framework/lifecycle/upgrade.go]: e2elog

File: test/e2e/framework/lifecycle/upgrade.go [hunk 3]
```diff
@@ -26,39 +26,41 @@ import (
...
 func RealVersion(s string) (string, error) {
-	Logf("Getting real version for %q", s)
-	v, _, err := RunCmd(path.Join(TestContext.RepoRoot, "hack/get-build.sh"), "-v", s)
+	e2elog.Logf("Getting real version for %q", s)
+	v, _, err := framework.RunCmd(path.Join(framework.TestContext.RepoRoot, "hack/get-build.sh"), "-v", s)
 	if err != nil {
 		return v, fmt.Errorf("error getting real version for %q: %v", s, err)
 	}
-	Logf("Version for %q is %q", s, v)
+	framework.Logf("Version for %q is %q", s, v)
```
Comment (andrewsykim) [inline test/e2e/framework/lifecycle/upgrade.go]: e2elog

File: test/e2e/lifecycle/cluster_upgrade.go [hunk 4]
```diff
@@ -32,6 +32,7 @@ import (
 	"k8s.io/kubernetes/test/e2e/chaosmonkey"
 	"k8s.io/kubernetes/test/e2e/framework"
 	"k8s.io/kubernetes/test/e2e/framework/ginkgowrapper"
+	"k8s.io/kubernetes/test/e2e/framework/lifecycle"
```
Comment (andrewsykim) [inline test/e2e/lifecycle/cluster_upgrade.go]: import alias e2elifecycle

**Comment by andrewsykim** [comment]
After reading through PR again, I think a better name for `test/e2e/framework/lifecycle/upgrade.go` would actually be something like `component_versions.go` or `versions.go`. Minor preference though. I'll defer to Tim for that one.

/assign @timothysc 
/lgtm

**Review by timothysc — APPROVED** [review]
/approve

**Pull request merged**